src/lib/storage/chunk.cpp

```diff
@@ -19,20 +19,20 @@ void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) { _columns.push_ba
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
   DebugAssert(values.size() == column_count(),
               "The number of passed arguments doesn't match the number of columns in the chunk.");
-  for (ColumnID value_index = (ColumnID)0; value_index < (ColumnID)values.size(); ++value_index)
-    _columns[value_index].get()->append(values[value_index]);
+  for (auto value_index = ColumnID{0}; value_index < ColumnID{values.size()}; ++value_index)
+    _columns[value_index]->append(values[value_index]);
 }
 
 std::shared_ptr<BaseSegment> Chunk::get_segment(ColumnID column_id) const {
-  if (column_id > column_count()) return nullptr;
+  DebugAssert(column_id < column_count(), "The given column_id is outside of the segment's columns.");
```

Comment:
Consider not using a `DebugAssert` here in order to also assure this in the release build.

Reply:
Actually we introduced the DebugAssert on purpose in order to have the following line fail loudly (because of vector access at an out-of-bounds location) in Release builds.